src/rgw/rgw_common.cc

```diff
@@ -196,6 +197,36 @@ is_err() const
   return !(http_ret >= 200 && http_ret <= 399);
 }
 
+
+// produces a string that interpets flags, such as "VERSIONED &
+// ~suspended & ~versions_suspended & ~datasync_disabled &
+// ~mfa_enabled & ~obj_lock_enabled"
+//
+// the vector pairs each flag mask with its desciptione
+//
+// if include_missing is true, includes flags with zero values in the
+// interpretation as well
+std::string flag_summary(std::vector<std::pair<int,std::string>> flag_list,
```

Comment:
the list of flag values/names should always be known at compile time, so we shouldn't need to allocate/copy any vectors or strings. you can replace `vector` -> `initializer_list` and `string` -> `string_view`